model.json

```diff
@@ -61,6 +61,37 @@
             "füge {amount} {unit} {ingredient} hinzu"
           ]
         },
+        {
+          "name": "AddIngredientRecipeIntent",
+          "slots": [
+            {
+              "name": "amount",
+              "type": "AMAZON.NUMBER",
+              "samples": [
+                "{amount} {unit} "
+              ]
+            },
+            {
+              "name": "unit",
+              "type": "ingredientUnit",
+              "samples": [
+                "{unit}"
+              ]
+            },
+            {
+              "name": "ingredient",
+              "type": "AMAZON.Food",
+              "samples": [
+                "Die Zutat ist {ingredient}",
+                "{ingredient}"
+              ]
+            }
+          ],
+          "samples": [
+            "füge {amount} {unit} {ingredient} zu Rezept dazu",
```

Comment:
Wäre eigentlich dafür hier nicht extra einen neuen Intent zu machen. Verwenden wir doch den `AddIngredientIntent`, den wir schon haben, aber unterscheiden dann nur zwischen den Kontexten.

Reply:
@bastidest2: in den AddIngredientIntent kann ich es einbauen.
Nur alles über den EditRecipeIntent zu machen find ich nicht gut.
@hmmst was sagst du?

Reply:
Mir ist es egal wo. Man muss nur beachten dass Intents (eigentlich nur eine "Phrase") nicht direkt auf einen eindeutigen Befehl mappen. Mehrere Handler müssen sich die Phrase anschauen und entscheiden ob sie im aktuellen Kontext verwendet werden kann.

Reply:
ja als wir das mit den Kontexten besprochen haben, haben wir eigentlich gesagt, dass es vom Intent/Befehl her genau das gleiche ist, ob man Zutaten zum Rezept oder zur Einkaufsliste hinzufügt...

@akramer123 wie würdest du es ohne den EditRecipeIntent machen? ich denke es ist sehr verwirrend, wenn man sagen muss: "für fünfzig Gramm Tomaten zum Rezept Tomatensuppe hinzu" oder? da ist es doch angenehmer, wenn man sagt "Bearbeite Rezept Tomatensuppe" "Füge fünfzig Gramm Tomaten hinzu".. und dann kann man noch weitere Dinge machen, bis man den "Bearbeitungsmodus" verlässt...

Reply:
vor allem ist momentan das Problem, wenn ich im Bearbeiten bin und sage "füge fünfzig gramm tomaten hinzu", fügt alexa sie zur Einkaufsliste..

Reply:
Also bei mir crasht es dann, weil kein Handler gefunden wird. Der normale Ingredients Handler handelt nämlich jetzt nur im Context "null".

Reply:
@hmmst 
Der EditRecipeIntent würde dann nur den Context bereitstellen.
Die anderen Handler könnten somit mit Überprüfung auf den Context unterschiedliche Aktionen durchführen. Zum Beispiel AddIngredientIntent fügt der Shoppinglist hinzu wenn kein rezept in Context und fügt dem Rezept im Context hinzu wenn eines vorhanden ist.
Man müsste somit nur den Code aus dem neuen Handler in den AddIngredientIntent verschieben und dort auf den Context prüfen.
Die Befehle wären somit also die selben.

Reply:
> vor allem ist momentan das Problem, wenn ich im Bearbeiten bin und sage "füge fünfzig gramm tomaten hinzu", fügt alexa sie zur Einkaufsliste..

Ja sollte eigentlich crashen.
Es muss wie oben beschrieben, der code verschoben werden und die canHandle Methode angepasst werden im AddIngredientHandler

Reply:
@akramer123 ja genau das meinte ich.. also ein normales "füge .. hinzu" fügt es zur Einkaufsliste und wenn man im EditRecipe Context ist, würde der gleiche Befehl die Zutat zum Rezept hinzufügen..:+1: